### PR TITLE
IGNITE-16045 Add documentation pages for the snapshot thread pool size property

### PR DESCRIPTION
**Pull request by Mmuzaf**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

**Comment by Mmuzaf** [comment]
@Nikita-tech-writer Can you review the patch?

File: docs/_docs/snapshots/snapshots.adoc [hunk 1]
```diff
@@ -76,16 +76,25 @@ snapshot data spread across the cluster. Each node keeps a segment of the snapsh
...
 
 See the link:persistence/snapshot-directory#configuring-snapshot-directory[Configuring Snapshot Directory] page for
 configuration examples.
 
+=== Snapshot Execution Pool
+
+By default, the snapshot thread pool size has value `4`. Decreasing the number of threads involving into snapshot creation will
+increase the total amount of time the snapshot being taking, however, this will keep the disk load within reasonable limits.
+
+See the link:perf-and-troubleshooting/thread-pools-tuning[Ignite Snapshot Execution Pool] page.
```
Comment (Nikita-tech-writer) [inline docs/_docs/snapshots/snapshots.adoc:96]:
```suggestion
See the link:perf-and-troubleshooting/thread-pools-tuning[Ignite Snapshot Execution Pool,window=_blank] page for more details.
```

File: docs/_docs/snapshots/snapshots.adoc [hunk 2]
```diff
@@ -76,16 +76,25 @@ snapshot data spread across the cluster. Each node keeps a segment of the snapsh
...
+already taken by the persistence files and can affect your application's performance by sharing the disk I/O with the
 Ignite Persistence routines, it's suggested to store the snapshot and persistence files on different media.
 
 See the link:persistence/snapshot-directory#configuring-snapshot-directory[Configuring Snapshot Directory] page for
 configuration examples.
 
+=== Snapshot Execution Pool
+
+By default, the snapshot thread pool size has value `4`. Decreasing the number of threads involving into snapshot creation will
+increase the total amount of time the snapshot being taking, however, this will keep the disk load within reasonable limits.
```
Comment (Nikita-tech-writer) [inline docs/_docs/snapshots/snapshots.adoc:94]:
```suggestion
By default, the snapshot thread pool size has a value of `4`. Decreasing the number of threads involved in the snapshot creation process 
increases the total amount of time for taking a snapshot. However, this keeps the disk load within reasonable limits.
```

**Comment by Nikita-tech-writer** [comment]
@Mmuzaf I've left a couple of suggestions. The rest LGTM! Thank you.

**Pull request merged**